packages/orchestration/src/contracts/stakingAccountHolder.js

```diff
@@ -0,0 +1,197 @@
+      helper: {
+        /** @throws if this holder no longer owns the account */
+        owned() {
+          const { account } = this.state;
+          if (!account) {
+            throw Fail`Using account holder after transfer`;
+          }
+          return account;
+        },
+        getUpdater() {
+          return this.state.topicKit.recorder;
+        },
+        /**
+         * _Assumes users has already sent funds to their ICA, until #9193
+         * @param {string} validatorAddress
+         * @param {Amount<'nat'>} ertpAmount
+         */
+        async delegate(validatorAddress, ertpAmount) {
+          // FIXME get values from proposal or args
+          // FIXME brand handling and amount scaling
+          const amount = {
+            amount: String(ertpAmount.value),
+            denom: 'uatom',
+          };
+
+          const account = this.facets.helper.owned();
+          trace('chainAccount', account);
+
+          // const delegatorAddress = await E(chainAccount).getAccountAddress();
+          const delegatorAddress = this.state.chainAddress;
+          trace('delegatorAddress', delegatorAddress);
+
+          const result = await E(account).executeEncodedTx([
+            txToBase64(
+              MsgDelegate.toProtoMsg({
+                delegatorAddress,
+                validatorAddress,
+                amount,
+              }),
+            ),
+          ]);
+          trace('Result', result);
+          const resultOrError = decodeTxResult(result, MsgDelegateResponse);
+          trace('Decoded Result', result);
+
+          if (JSON.stringify(resultOrError) === '{}') return 'Success';
+          throw Fail`Failed to delegate. ${resultOrError}`;
+        },
+      },
+      invitationMakers: {
+        Delegate(validatorAddress, amount) {
+          return this.facets.holder.makeDelegateInvitation(
+            validatorAddress,
+            amount,
+          );
+        },
+        CloseAccount() {
+          return this.facets.holder.makeCloseAccountInvitation();
+        },
+        TransferAccount() {
+          return this.facets.holder.makeTransferAccountInvitation();
+        },
+      },
+      holder: {
+        getPublicTopics() {
+          const { topicKit } = this.state;
+          return harden({
+            account: {
+              description: PUBLIC_TOPICS.account[0],
+              subscriber: topicKit.subscriber,
+              storagePath: topicKit.recorder.getStoragePath(),
+            },
+          });
+        },
+        /**
+         *
+         * @param {string} validatorAddress
+         * @param {Amount<'nat'>} ertpAmount
+         */
+        async delegate(validatorAddress, ertpAmount) {
+          trace('delegate', validatorAddress, ertpAmount);
+          return this.facets.helper.delegate(validatorAddress, ertpAmount);
+        },
+        /** @param {bigint} value */
+        makeAtomAmount(value) {
+          return AmountMath.make(zcf.getTerms().brands.ATOM, value);
```

Comment:
Only here because I wasn't sure how to make an amount that EV likes: https://github.com/Agoric/agoric-sdk/pull/9195/files#diff-591b38c7c82b1d2e45f9fd7ad844cd799e8242267e5290d8b7de8fe029934994R125

Reply:
fixed in fda336a

Reply:
fixed in [fda336a](https://github.com/Agoric/agoric-sdk/pull/9195/commits/fda336aefef52f5d6c70c1f2c30bb158d1525fe2)